projects/components/src/filtering/filter-bar/filter-chip/filter-chip.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable } from '@angular/core';
+import { FilterBuilderLookupService } from '../../filter/builder/filter-builder-lookup.service';
+import { AbstractFilterBuilder } from '../../filter/builder/types/abstract-filter-builder';
+import { IncompleteFilter } from '../../filter/filter';
+import { FilterAttribute } from '../../filter/filter-attribute';
+import { FilterOperator } from '../../filter/filter-operators';
+import { FilterParserLookupService } from '../../filter/parser/filter-parser-lookup.service';
+import { SplitFilter } from '../../filter/parser/parsed-filter';
+import { AbstractFilterParser, splitFilterStringByOperator } from '../../filter/parser/types/abstract-filter-parser';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FilterChipService {
+  public constructor(
+    private readonly filterBuilderLookupService: FilterBuilderLookupService,
+    private readonly filterParserLookupService: FilterParserLookupService
+  ) {}
+
+  public autocompleteFilters(attributes: FilterAttribute[], text: string = ''): IncompleteFilter[] {
+    return attributes
+      .filter(attribute => this.filterBuilderLookupService.isBuildableType(attribute.type))
+      .flatMap(attribute => this.autocompleteFilter(attribute, text))
+      .filter(incompleteFilter => this.filterMatchesUserText(text, incompleteFilter));
+  }
+
+  private filterMatchesUserText(text: string, incompleteFilter: IncompleteFilter): boolean {
+    const isStringMatch = incompleteFilter.userString.toLowerCase().includes(text.trim().toLowerCase());
```

Comment:
should this be `startsWith` instead of `includes`?

Reply:
`include` works better here. If we have two attributes like `Service Name` and `API Name`, we want a user to be able to type "name" and see both options. They can then tab to autofill from the options.